### ISSUE TITLE
About the residual strategy

### ISSUE THREAD

**Issue by JasonBourneZYX** [SaoYan/DnCNN-PyTorch#12]
Dear author, 
I `ve found a difference between your train code and CSZN DNCNN pytorch train code.

1.  In his train code, the loss is calculated like this:

` loss = criterion(model(batch_y), batch_x)`_
Since the output of his model was clean image. He define the loss by output and clean-groundtruth. 

2. But in ur train code, the loss is calculated like this:
`loss = criterion(out_train, noise) / (imgn_train.size()[0]*2)`
Since the output of ur model was the residual noise map. You define the loss by output and the noise-groundtruth

So the question is: 

- Would this difference influence the training result? 
- For denoising, is it maybe a better way to take clean image as ground-truth instead of noise residual?

**Comment by liusasasasa** [comment]
> 亲爱的作者，我发现您的火车代码和 CSZN DNCNN pytorch 火车代码之间存在差异。
> 
> 1. 在他的火车代码中，损失是这样计算的：
> 
> ` loss = criterion(model(batch_y), batch_x)`_由于他的模型的输出是干净的图像。他通过输出和干净的地面真实来定义损失。
> 
> 2. 但是在训练代码中，损失是这样计算的：因为模型的输出是残余噪声图。您可以通过输出和噪声-地面真值来定义损耗`loss = criterion(out_train, noise) / (imgn_train.size()[0]*2)`
> 
> 所以问题是：
> 
> * 这种差异会影响训练结果吗？
> * 对于降噪，将干净的图像作为地面真实而不是噪声残留是否是一种更好的方法？

hello,do you understand now?
